pkg/apicapi/apicapi.go

```diff
@@ -746,6 +747,61 @@ func (conn *ApicConnection) logErrorResp(message string, resp *http.Response) {
 	}
 }
 
+// To make sure cluster's POD/NodeBDs and L3OUT are all mapped
+// to same and correct VRF.
+func (conn *ApicConnection) ValidateAciVrfAssociation(acivrfdn string, expectedVrfRelations []string) error {
+    var aciVrfBdL3OuttDns []string
+	args := []string{
+		"query-target=subtree",
+		"target-subtree-class=fvRtCtx,fvRtEctx",
+	}
+
+	uri := fmt.Sprintf("/api/mo/%s.json?%s", acivrfdn, strings.Join(args, "&"))
+	url := fmt.Sprintf("https://%s%s", conn.apic[conn.apicIndex], uri)
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		conn.log.Error("Could not create request: ", err)
+		return err
+	}
+	conn.sign(req, uri, nil)
+	resp, err := conn.client.Do(req)
+	if err != nil {
+		conn.log.Error("Could not get subtree for ", acivrfdn, ": ", err)
+		return err
+	}
+	defer complete(resp)
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
+		conn.logErrorResp("Could not get subtree for "+acivrfdn, resp)
+		return err
+	}
+
+	var apicresp ApicResponse
+	err = json.NewDecoder(resp.Body).Decode(&apicresp)
+	if err != nil {
+		conn.log.Error("Could not parse APIC response: ", err)
+		return err
+	}
+
+    for _, obj := range apicresp.Imdata {
+        for _, body := range obj {
+			tDn,ok := body.Attributes["tDn"].(string)
+			if !ok {
+				continue
+			}
+			aciVrfBdL3OuttDns = append(aciVrfBdL3OuttDns, tDn)
```

Comment:
can you check if you can move the below for loop to here

Reply:
@jayaramsatya I don't like too many loops as well. In this case, APIC can always have a superset of tDns and hence it's not possible to compare and error out. We can do it by having a map of expectedDns and marking them for each tDn that we read from APIC, but it gets unreadable. Given that we are only dealing with handful (3 as of now) of expectedDns, it's ok to have that additional loop in the end.